ambry-router/src/main/java/com/github/ambry/router/TtlUpdateManager.java

```diff
@@ -81,42 +83,64 @@ class TtlUpdateManager {
 
   /**
    * Submits {@link TtlUpdateOperation}(s) to this {@link TtlUpdateManager}.
-   * @param blobIdStrs The original blobId strings
+   * @param blobIdStr The blobId of the simple blob or the metadata blob in case of composite blob.
+   * @param chunkIdStrs The blob ids of the metadata blob's chunks.
    * @param serviceId The service ID of the service updating the ttl of the blob(s). This can be null if unknown.
    * @param expiresAtMs The new expiry time (in ms) of the blob.
    * @param futureResult The {@link FutureResult} that will contain the result eventually and exception if any.
    * @param callback The {@link Callback} that will be called on completion of the request.
    * @param quotaChargeCallback {@link QuotaChargeCallback} object to account for quota.
    * @throws RouterException if the blobIdStr is invalid.
    */
-  void submitTtlUpdateOperation(Collection<String> blobIdStrs, String serviceId, long expiresAtMs,
+  void submitTtlUpdateOperation(String blobIdStr, Collection<String> chunkIdStrs, String serviceId, long expiresAtMs,
       FutureResult<Void> futureResult, Callback<Void> callback, QuotaChargeCallback quotaChargeCallback)
       throws RouterException {
-    List<BlobId> blobIds = new ArrayList<>();
-    for (String blobIdStr : blobIdStrs) {
-      BlobId blobId = RouterUtils.getBlobIdFromString(blobIdStr, clusterMap);
-      if (blobId.getDatacenterId() != ClusterMap.UNKNOWN_DATACENTER_ID
-          && blobId.getDatacenterId() != clusterMap.getLocalDatacenterId()) {
-        routerMetrics.ttlUpdateBlobNotOriginateLocalOperationRate.mark();
-      }
-      blobIds.add(blobId);
+    BlobId blobId = RouterUtils.getBlobIdFromString(blobIdStr, clusterMap);
+    if (blobId.getDatacenterId() != ClusterMap.UNKNOWN_DATACENTER_ID
+        && blobId.getDatacenterId() != clusterMap.getLocalDatacenterId()) {
```

Comment:
please convert to a helper like isRemoteDataCenter(). Too many booleans make it hard to read in if-clause

Reply:
done